File: Day348/ConvertAnArrayInto2DArrayWithConditions.go
func findMatrix(nums []int) [][]int {
    temp := make([]int, len(nums)+1)
    answer := make([][]int, 0)
    for _,v := range nums {
        if temp[v] >= len(answer) {
            answer = append(answer, make([]int, 0))
        }
        answer[temp[v]] = append(answer[temp[v]], v)
        temp[v]++
    }
    return answer
}

File: Day348/ConvertAnArrayInto2DArrayWithConditions.java
class Solution {
    public List<List<Integer>> findMatrix(int[] nums) {
        int temp[] = new int[nums.length+1];
        List<List<Integer>> answer = new ArrayList<>();
        for (int v: nums) {
            if (temp[v] >= answer.size()) {
                answer.add(new ArrayList<>());
            }
            answer.get(temp[v]).add(v);
            temp[v]++;
        }
        return answer;
    }

}
